django/thunderstore/api/cyberstorm/views/package_detail.py

```diff
@@ -0,0 +1,152 @@
+from thunderstore.repository.models.package_version import PackageVersion
+
+
+class DependencySerializer(serializers.Serializer):
+    """
+    Dependencies of a given PackageVersion, listed in a given Community.
+
+    community_identifier and namespace are not present by default and
+    need to be annotated to the object.
+    """
+
+    community_identifier = serializers.CharField()
+    description = serializers.CharField()
+    icon_url = serializers.CharField(source="icon.url")
+    name = serializers.CharField()
+    namespace = serializers.CharField(source="package.namespace.name")
+    version_number = serializers.CharField()
+
+
+class TeamSerializer(serializers.Serializer):
+    """
+    Minimal information to present the team on package detail view.
+    """
+
+    name = serializers.CharField()
+    members = CyberstormTeamMemberSerializer(many=True)
+
+
+class ResponseSerializer(serializers.Serializer):
+    """
+    Data shown on package detail view.
+
+    Expects an annotated and customized CustomListing object.
+    """
+
+    categories = CyberstormPackageCategorySerializer(many=True)
+    community_identifier = serializers.CharField(source="community.identifier")
+    community_name = serializers.CharField(source="community.name")
+    datetime_created = serializers.DateTimeField(source="package.latest.date_created")
+    dependant_count = serializers.IntegerField(min_value=0)
+    dependencies = DependencySerializer(many=True)
+    description = serializers.CharField(source="package.latest.description")
+    download_count = serializers.IntegerField(min_value=0)
+    download_url = serializers.CharField(source="package.latest.full_download_url")
+    full_version_name = serializers.CharField(source="package.latest")
+    has_changelog = serializers.SerializerMethodField()
+    icon_url = serializers.CharField(source="package.latest.icon.url")
+    install_url = serializers.CharField(source="package.latest.install_url")
+    is_deprecated = serializers.BooleanField(source="package.is_deprecated")
+    is_nsfw = serializers.BooleanField(source="has_nsfw_content")
+    is_pinned = serializers.BooleanField(source="package.is_pinned")
+    last_updated = serializers.DateTimeField(source="package.date_updated")
+    latest_version_number = serializers.CharField(
+        source="package.latest.version_number",
+    )
+    name = serializers.CharField(source="package.name")
+    namespace = serializers.CharField(source="package.namespace.name")
+    rating_count = serializers.IntegerField(min_value=0)
+    size = serializers.IntegerField(min_value=0, source="package.latest.file_size")
+    team = TeamSerializer(source="package.owner")
+    website_url = serializers.CharField(source="package.latest.website_url")
+
+    def get_has_changelog(self, listing: PackageListing) -> bool:
+        changelog = listing.package.latest.changelog
+        return False if changelog is None else bool(changelog.strip())
```

Comment:
Any reason this can't be annotated in the QS?

Not a blocker

Reply:
Performance primarily, serializer methods are slower than annotations from my experience (and it can make a pretty big difference with larger sets of data)

Reply:
Addressed in https://github.com/thunderstore-io/Thunderstore/pull/969